starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,151 @@
+from copy import deepcopy
+from typing import Callable, Iterable, Optional, Union, Sequence
+
+import numpy as np
+
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.types import Axes, Clip, Coordinates
+from starfish.core.util import click
+from starfish.core.util.dtype import preserve_float_range
+from ._base import FilterAlgorithmBase
+
+
+class Reduce(FilterAlgorithmBase):
+    """
+    Reduces the dimensions of the ImageStack by applying a function
+    along one or more axes.
+
+    Parameters
+    ----------
+    dims : Axes
+        one or more Axes to project over
+    func : Union[str, Callable]
+        function to apply across the dimension(s) specified by dims.
+        If a function is provided, it should follow the form specified by
+        DataArray.reduce():
+        http://xarray.pydata.org/en/stable/generated/xarray.DataArray.reduce.html
+
+        The following strings are valid:
+            max: maximum intensity projection (applies numpy.amax)
+            mean: take the mean across the dim(s) (applies numpy.mean)
+            sum: sum across the dim(s) (applies numpy.sum)
+
+    clip_method : Union[str, Clip]
+        (Default Clip.CLIP) Controls the way that data are scaled to retain skimage dtype
+        requirements that float data fall in [0, 1].
+        Clip.CLIP: data above 1 are set to 1, and below 0 are set to 0
+        Clip.SCALE_BY_IMAGE: data above 1 are scaled by the maximum value, with the maximum
+        value calculated over the entire ImageStack
+        Clip.SCALE_BY_CHUNK: data above 1 are scaled by the maximum value, with the maximum
+        value calculated over each slice, where slice shapes are determined by the group_by
+        parameters
+
+    See Also
+    --------
+    starfish.types.Axes
+
+    """
+
+    def __init__(
+        self, dims: Iterable[Union[Axes, str]], func: Union[str, Callable] = 'max',
+        clip_method: Union[str, Clip] = Clip.CLIP
+    ) -> None:
+
+        self.dims = dims
+        self.clip_method = clip_method
+
+        # If the user provided a string, convert to callable
+        if isinstance(func, str):
+            if func == 'max':
+                func = 'amax'
+            func = getattr(np, func)
+        self.func = func
+
+    _DEFAULT_TESTING_PARAMETERS = {"dims": ['r'], "func": 'max'}
+
+    def run(
+            self,
+            stack: ImageStack,
+            in_place: bool = False,
```

Comment:
I would explicitly not support in_place.  It doesn't make any sense.

Reply:
Sounds good. In max_proj, the `run()` method still accepts an `in_place` argument and just doesn't do anything with it. Should I do the same here? Maybe give a warning?

Reply:
Just remove the in_place and n_processes arguments, as neither do what they are advertised to do.